agentlib/core/environment.py

```diff
@@ -43,137 +40,230 @@ class EnvironmentConfig(BaseModel):
-        self._config = EnvironmentConfig.model_validate(config)
-        self._create_env()
-
-    def _create_env(self):
-        """
-        Sets the runtime environment as discrete event dispatcher. We Use the
-        simpy package for this:
-        https://simpy.readthedocs.io/en/latest/contents.html
-
-        The environment is initialized according to configuration. If an
-        existing one is parsed it will only check if the local configuration
-        aligns with external configuration.
-
-        Args:
-            env(simpy.Environment): Environment where the local processes
-            should be added to
-
-        Returns:
-
-        """
-        if self.config.rt:
-            logger.info("Initializing real time runtime environment...")
-            simpy.RealtimeEnvironment.__init__(
-                self,
-                factor=self.config.factor,
-                initial_time=self.config.initial_time,
-                strict=self.config.strict,
-            )
-            self.pretty_time = self.rt_time
-        else:
-            logger.info("Initializing runtime environment...")
-            simpy.Environment.__init__(
-                self, **self.config.model_dump(include={"initial_time"})
-            )
-            self.pretty_time = self.sim_time
-        # regularily print sim time, if clock is on, and log level is sufficient
-        if self.config.clock and logger.level <= logging.INFO:
+    @property
+    def offset(self) -> EnvironmentConfig:
+        """Start time offset of the environment."""
+        return self.config.offset
+
+    @property
+    def time(self) -> float:
+        """Get the current time of the environment.
+        If RT is enabled, the unix-time is returned."""
+        return self.now
+
+    def clock(self):
+        """Define a clock loop to increase the now-timer every other second
+        (Or whatever t_sample is)"""
+        # if log level is not info or debug, this process can terminate
+        while True:
+            logger.info("Current simulation time: %s", self.pretty_time())
+            yield self.timeout(self.config.t_sample)
+
+    def pretty_time(self):
+        ...
+
+
+class InstantEnvironment(CustomSimpyEnvironment):
+    """A customized version of the simpy environment. Handles execution of modules
+    processes and manages time for instant execution mode."""
+
+    def __init__(self, *, config: EnvironmentConfig):
+        super().__init__(initial_time=config.offset)
+        self._config = config
+        if self.config.clock:
             self.process(self.clock())
 
-    def step(self) -> None:
-        if self.config.rt:
-            simpy.RealtimeEnvironment.step(self)
+    def pretty_time(self) -> str:
+        """Returns the time in a nice format. Datetime if realtime, seconds if not
+        realtime. Implemented as rt_time or sim_time"""
+        return f"{self.time:.2f}s"
+
+
+class RealtimeEnvironment(simpy.RealtimeEnvironment, CustomSimpyEnvironment):
```

Comment:
You now have a class which inherits both Realtime and standard Environment. Does this lead to weird behaviour? Before it was only RT

Reply:
I checked the __mro__ during debugging, I think it will be fine.